akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala

```diff
@@ -24,6 +26,10 @@ private[akka] trait ReceiveTimeout { this: ActorCell ⇒
 
   final def setReceiveTimeout(timeout: Duration): Unit = receiveTimeoutData = receiveTimeoutData.copy(_1 = timeout)
 
+  protected def checkReceiveTimeoutIfNeeded(message: Any): Unit =
+    if (hasTimeoutData)
+      checkReceiveTimeout(!message.isInstanceOf[NotInfluenceReceiveTimeout])
+
   final def checkReceiveTimeout(reschedule: Boolean = true): Unit = {
     val (recvtimeout, task) = receiveTimeoutData
     recvtimeout match {
```

Comment:
We have had a few issues with receive timeout so we must be sure that it's doing the same, see below comment... as far as I can see it's correct.

Reply:
Yep, needs proper review. I tried to make sure not to break something.